pymongo/asynchronous/mongo_client.py

```diff
@@ -1040,6 +1041,23 @@ async def target() -> bool:
         self._kill_cursors_executor = executor
         self._opened = False
 
+    def append_metadata(self, driver_info: DriverInfo) -> None:
+        """
+        Appends the given metadata to existing driver metadata.
+        """
+        metadata = self._options.pool_options.metadata
+        for k, v in driver_info._asdict().items():
+            if v is None:
+                continue
+            if k in metadata:
+                metadata[k] = f"{metadata[k]}|{v}"
+            elif k in metadata["driver"]:
+                metadata["driver"][k] = "{}|{}".format(
```

Comment:
Could we move this logic into PoolOptions and refactor it with the existing code?:
```
        if driver:
            if driver.name:
                self.__metadata["driver"]["name"] = "{}|{}".format(
                    self.__metadata["driver"]["name"],
                    driver.name,
                )
            if driver.version:
                self.__metadata["driver"]["version"] = "{}|{}".format(
                    _METADATA["driver"]["version"],
                    driver.version,
                )
            if driver.platform:
                self.__metadata["platform"] = "{}|{}".format(_METADATA["platform"], driver.platform)
```